src/storage/txn/scheduler.rs

```diff
@@ -527,25 +548,34 @@ impl<E: Engine, L: LockManager, P: PdClient + 'static> Scheduler<E, L, P> {
 
     /// Delivers a command to a worker thread for processing.
     fn process_by_worker(self, snapshot: E::Snap, task: Task) {
+        let timer = Instant::now_coarse();
         let tag = task.cmd.tag();
         SCHED_STAGE_COUNTER_VEC.get(tag).process.inc();
 
         self.get_sched_pool(task.cmd.priority())
             .clone()
             .pool
             .spawn(async move {
+                SCHED_PRE_HANDLE_3_DURATIONS_VEC
+                    .get(tag)
+                    .observe(timer.elapsed_secs());
+                let mut timer = Instant::now_coarse();
+
                 fail_point!("scheduler_async_snapshot_finish");
 
                 let read_duration = Instant::now_coarse();
 
                 let region_id = task.cmd.ctx().get_region_id();
                 let ts = task.cmd.ts();
-                let timer = Instant::now_coarse();
                 let mut statistics = Statistics::default();
 
                 if task.cmd.readonly() {
                     self.process_read(snapshot, task, &mut statistics);
                 } else {
+                    SCHED_PRE_HANDLE_2_DURATIONS_VEC
```

Comment:
`SCHED_PRE_HANDLE_2_DURATIONS_VEC` seems no much different from `SCHED_PRE_HANDLE_3_DURATIONS_VEC`

Reply:
SCHED_PRE_HANDLE_3_DURATIONS_VEC was intended to record the accurate duration of in-queue time, but as you said the SCHED_PRE_HANDLE_2_DURATIONS_VEC  seems not much time, so I will merge then